### ISSUE TITLE
Custom Column in Models doesn't respect filtering when binning is applied

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#26202]
**Describe the bug**
When you create a model having a custom column then you proceed to create a question with this model having filters and binning by this custom column the binning strategy applied doesn't respect the filter and uses the full data within the custom column.

**To Reproduce**
1. Go to New -> Question -> Orders -> Custom Column ( Field Formula =[Total] * 1 and name = custom_tax) -> Save -> Turn into model 
2. Go to Metadata -> Custom_Column -> Set field_type to Quantity
4. Go to New > Question -> Model -> Filter Custom_Column > 20 and < 40 -> Group by custom_column (10 bins) and count -> Result only 1 bin

<img width="1290" alt="image" src="https://user-images.githubusercontent.com/110378427/199294155-274df157-da22-4496-b46a-fe047f33f228.png">

**Expected behavior**
Since for this bin there are more than 2000 values I would expect it to bin these values in 10 or at least some kind of binning.

**Information about your Metabase Installation:**

Version 42 - 44.5 

**Additional context**

If you save the question and want to investigate further, you can get the question metadata by going to the URL /api/card/123, where 123 is the question ID. You can notice that in the binning part it's taking the whole column data and generating the binnig structure on that rather than the result of the filters (min_value should be 20 and max_value should be 40):

<img width="741" alt="image" src="https://user-images.githubusercontent.com/110378427/199295477-f69506dc-8da1-48ac-b5b4-ef003a001b23.png">

**Comment by flamber** [comment]
Likely very related to #9321/#12004/#12210

**Comment by camsaul** [comment]
I think the problem here is that we've never supported binning on custom columns, so it probably shouldn't have let you set those options in the first place, because they don't actually have any effect at this point in time

**Comment by camsaul** [comment]
Closing as duplicate of #11371: binning is not currently supported for custom columns. Hopefully we can add support for this in the future, it should be easy once #28689 is in place.

**Comment by uladzimirdev** [comment]
As I see, it works for binning `> 10`, but 10 and auto create a single column which looks wrong. 90% it's a BE issue

**Comment by camsaul** [comment]
The QP code for binning introspects the query and looks at filters that might affect the min/max -- and it seems to work for normal Tables. We should figure out why it's not working for Models and make it work there too -- probably not super difficult. Not sure if this needs to be done in the MLv2 drill thru or in QP binning middleware
